client/src/components/subscription/subscriptionForm.js

```diff
@@ -0,0 +1,239 @@
+import React from 'react';
+import { toast } from 'react-toastify';
+import ReactDayPicker from '../date/datepicker';
+import { convertWeekdaysToDates } from '../../utils/frontendDateUtils';
+
+function convertStringToDate(datesArray) {
+  return datesArray.map((day) => new Date(day));   
+};
+
+class SubscriptionForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { method, prevSubscription } = props;
+    let parseDate = prevSubscription ? prevSubscription.dueDate.dates : null;
```

Comment:
you can use optional chaining here instead

```suggestion
    let parseDate = prevSubscription?.dueDate?.dates || null;
```